### DISCUSSION TITLE
Is it possible to add an object from stl file to MeshcatVisualizer?

### DISCUSSION THREAD

**Discussion by eirtech** [stack-of-tasks/pinocchio#2015] [Q&A]
I want to load an object from stl file I have. I already tried to make a geometry object and also use the load_mesh function. But it failed. Would you please give me a hint how to do it?

**Comment by nim65s** [comment]
Hi,

Did you look into the `examples` directory ?
We have:
- https://github.com/stack-of-tasks/pinocchio/blob/master/examples/display-shapes-meshcat.py
- https://github.com/stack-of-tasks/pinocchio/blob/master/examples/meshcat-viewer.py
- https://github.com/stack-of-tasks/pinocchio/blob/master/examples/meshcat-viewer-dae.py

**Reply by eirtech** [reply]
Hi,

Thanks for your answer. Yes, I am able to add a `URDF` model, and also geometry object which is defined by `hppfcl`, but I can't load a `.stl ` mesh directly to the scene. I tried to construct a geometry object from that `.stl` file I have, but it was not successful.

**Reply by nim65s** [reply]
Ok, then a quick workaround is to make a URDF with only your single STL.

I think we can provide an example to load directly an STL without going through URDF, but I personnaly won't have time to double check this today

**Reply by eirtech** [reply]
I wanted to say that I did used another way to do it. I just made a small `URDF` with that `.stl` mesh. and then load the `URDF`. I thought probably I could use something like `meshcat.geometry.StlMeshGeometry` to make and load an object directly, but at least I was not successful with that.

Thanks again for your response.
